File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/EventNotificationSample/src/com/hp/workpath/sample/eventnotificationsample/task/ConfigInitializationTask.java
package com.hp.workpath.sample.eventnotificationsample.task;

import android.content.Context;

import com.hp.workpath.api.config.ConfigService;
import com.hp.workpath.sample.eventnotificationsample.R;

public class ConfigInitializationTask extends InitializationTask {
    public ConfigInitializationTask(Context context, ResultHandler handler) {
        super(context, handler);
    }

    @Override
    public boolean isSupported(Context context) {
        return ConfigService.isSupported(context);
    }

    @Override
    public int getExceptionMessage() {
        return R.string.config_service_not_supported;
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/EventNotificationSample/src/com/hp/workpath/sample/eventnotificationsample/task/ConfigInitializationTask.kt
package com.hp.workpath.sample.eventnotificationsample.task

import android.content.Context
import com.hp.workpath.api.config.ConfigService
import com.hp.workpath.sample.eventnotificationsample.R

class ConfigInitializationTask(context: Context, handler: ResultHandler)
    : InitializationTask(context, handler) {
    override fun isSupported(context: Context?): Boolean {
        return ConfigService.isSupported(context?: return false)
    }

    override fun getExceptionMessage(): Int {
        return R.string.config_service_not_supported
    }
}